sys/net/gnrc/netreg/gnrc_netreg.c

```diff
@@ -95,7 +95,9 @@ static gnrc_netreg_entry_t *_netreg_lookup(gnrc_netreg_entry_t *from,
 
     if (from || !_INVALID_TYPE(type)) {
         gnrc_netreg_entry_t *head = (from) ? from->next : netreg[type];
-        LL_SEARCH_SCALAR(head, res, demux_ctx, demux_ctx);
+        if (head) {
+            LL_SEARCH_SCALAR(head, res, demux_ctx, demux_ctx);
```

Comment:
IIRC `LL_SEARCH_SCALAR` should cover an empty list (`head == NULL`). So what happened there?

Reply:
Hm you are right, this resolved to [`LL_FOREACH2`](https://github.com/RIOT-OS/RIOT/blob/master/sys/include/utlist.h#L473) which should not loop if `head` is NULL.

So something else must happen here.